### PR TITLE
Modernize React App

### PR DESCRIPTION
**Pull request by alecsci**
- replace react-scripts with vite setup
- upgrade dependencies
- separate dev dependencies from dependencies
- remove incomplete test setup (created separate issue)
- remove webvitals
- remove unused assets
- update readme
- enable eslint and prettier

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,53 +1,45 @@
...
-    "@types/react": "^17.0.2",
-    "@types/react-dom": "^17.0.1",
-    "axios": "^0.21.1",
-    "react": "^17.0.1",
-    "react-dom": "^17.0.1",
-    "react-loader-spinner": "^4.0.0",
-    "react-scripts": "^4.0.3",
-    "typescript": "^4.2.2",
-    "web-vitals": "^1.1.0"
+    "@vitejs/plugin-react": "^4.3.1",
```
Comment (Helias) [inline package.json:9]: I always recommend to put only FIXED version without `^` or `~`

File: src/components/shared/Util.ts [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
+const BRACKET_LEVEL_PREDICATES: { [key: number]: Predicate } = {
+  0: () => true, // no filter selected, show all levels
+  1: (player: Player) => player.level < 20, // show 1-19
+  2: (player: Player) => player.level >= 20 && player.level < 30,
+  3: (player: Player) => player.level >= 30 && player.level < 40,
+  4: (player: Player) => player.level >= 40 && player.level < 50,
+  5: (player: Player) => player.level >= 50 && player.level < 60,
+  6: (player: Player) => player.level >= 60 && player.level < 70,
+  7: (player: Player) => player.level >= 70 && player.level < 80,
+  8: (player: Player) => player.level === 80, // show 80 only
```
Comment (Helias) [inline src/components/shared/Util.ts:19]: nice

**Review by Helias — APPROVED** [review]

**Comment by alecsci** [comment]
@FrancescoBorzi or @Yehonal want to have a second look ?

**Comment by FrancescoBorzi** [comment]
Thanks for mentioning me but unfortunately I'm quite busy and cannot review this, sorry :(

I see Helias has already approved and you are the original author of this app so you can go ahead and merge this

**Pull request merged**